grug/jellyfish-merkle/src/tree.rs

```diff
@@ -375,45 +375,67 @@ impl<'a> MerkleTree<'a> {
         batch: Vec<(Hash256, Hash256)>,
         existing_leaf: Option<LeafNode>,
     ) -> StdResult<Outcome> {
-        let new_node = match (batch.len(), existing_leaf) {
-            (0, None) => {
-                return Ok(Outcome::Unchanged(None));
-            },
-            (0, Some(leaf_node)) => Node::Leaf(leaf_node),
+        match (batch.len(), existing_leaf) {
+            // The subtree to be created is empty: do nothing.
+            (0, None) => Ok(Outcome::Unchanged(None)),
+            // The subtree to be created contains exactly one node, which is an
+            // existing leaf node.
+            (0, Some(leaf_node)) => Ok(Outcome::Unchanged(Some(Node::Leaf(leaf_node)))),
+            // The subtree to be created contains exactly one node, which is a
+            // new leaf node.
+            // This case requires special attention: we don't save the node yet,
+            // because it the path may be collapsed if it's sibling gets deleted.
```

Comment:
```suggestion
            // because the path may be collapsed if it's sibling gets deleted.
```

Reply:
There's another typo in this sentence (its/it's)

Reply:
updated the comment